doc/development.md

```diff
@@ -0,0 +1,79 @@
+## Development documentation
```

Comment:
Similarlo to advanced_topics.md and architecture.md this document should be linked from the toplevel README.md.

Reply:
...toplevel README.md or maybe other

Reply:
Readme updated. Fix c45ba05